docs/notebooks/Word2Vec_Multistream.ipynb

```diff
@@ -0,0 +1,212 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Word2Vec Multistream API Tutorial\n",
+    "\n",
+    "This tutorial introduces new **`corpus_file`** argument for **`gensim.models.word2vec.Word2Vec`** model and how to use it. \n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "Because standard Word2Vec training with `sentences` argument doesn't scale so well when the number of workers is large, special `corpus_file` argument was added to tackle this problem. Training with `corpus_file` yields **significant performance boost** (training is 300% faster with 14 workers in comparison to training with `sentences` argument). Also, it outruns performance of original Word2Vec C tool in terms of words/sec processing speed.\n",
+    "\n",
+    "While providing such benefits in performance, `corpus_file` argument accepts path to your corpus file which must be in a format of `gensim.models.word2vec.LineSentence` (one sentence per line, words are separated by whitespaces).\n",
+    "\n",
+    "\n",
+    "**Note**: you have to build `gensim` with Cython optimizations (`FAST_VERSION >= 0`) in order to be able to use `corpus_file` argument."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## In this tutorial\n",
+    "\n",
+    "* We will show how to use the new API.\n",
+    "* We compare performance of `corpus_file` vs. `sentences` arguments on English Wikipedia.\n",
+    "* We will show that accuracies on `question-words.txt` are almost the same for both modes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Usage is really simple\n",
+    "\n",
+    "You only need:\n",
+    "\n",
+    "1. Save your corpus in LineSentence format (you may use `gensim.utils.save_as_line_sentence(your_corpus, your_corpus_file)` to save your corpus).\n",
+    "2. Change `sentences=your_corpus` argument to `corpus_file=your_corpus_file` in `Word2Vec.__init__`, `Word2Vec.build_vocab`, `Word2Vec.train` calls.\n",
```

Comment:
a little code example (with small corpus, like `text8`) is good idea here, something like
```python
corpus = api.load("text8")

save_as_line_sentence(corpus, "my_corpus.txt")

model = Word2Vec(corpus_file="my_corpus.txt", ...)
```

Reply:
For what reason? I'm already showing the example of usage, no?

Reply:
this is a good tone: me (as a user) open notebook and immediately see how to use it (small & simple code), if I need something else - I'll scroll down.

Reply:
👍 agree